### PR TITLE
Jiyuan added scheduled-on date to scheduled blue square reason

### PR DESCRIPTION
**Pull request by Jiyuanxia**
# Description
Screenshot 2023-11-29 153713
Screenshot 2023-11-29 153739

## Related PRS (if any):
This frontend PR is related to the [#640](https://github.com/OneCommunityGlobal/HGNRest/pull/640) backend PR.
To test this backend PR you need to checkout the [#640](https://github.com/OneCommunityGlobal/HGNRest/pull/640) backend PR.

## How to test:
1. check into current branch
2. `npm install`
3. `npm run start:local`
4. log in as admin
5. go to anyone's profile page
6. add a blue square to this person by clicking `+`
7. check if the new blue square contains the created date before the summary with right format(month/day) by hovering it
8. click this new blue square
9. check if  **created date** appears 

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92963742/2ead345b-2013-4f42-aeef-d8dbf69ca07b

## Note:
Old blue squares do not have **created date**

### CODE REVIEW

**Comment by Cavein254** [comment]
Hi @Jiyuanxia, reviewed your pr and everything works as expected :fire: 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13707896/cd66b179-9fa7-4f66-ac2f-90bcc2b43b24

**Review by this-journey — DISMISSED** [review]
Hello! I reviewed your PR and everything seems to be working perfectly.  I know this may be outside of the scope of your PR, however, may I suggest making the blue squares section a little bit bigger. In comparison to the "Schedule Blue Square Reson" button it's extremely small and difficult to find if you don't know exactly what you're looking for.  Great job! I've attached a video for reference.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/93569791/16b6c075-ae26-4418-b549-21ae3921ab48

**Review by Alforoan — DISMISSED** [review]
Assining blue square and hovering both worked. Nice!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/036b59f3-7dad-453e-ab45-5aef0dc29d59

**Comment by DragonSlayer2401** [comment]
I tested your PR and everything works as intended. I created the square, updated the square, and deleted the square on top of testing for the created date and hovering over it. Please watch the overview video below for more details:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102876103/2ad5520b-ee60-46b0-99dd-fbe1500d4542

**Review by keyunhuangg — DISMISSED** [review]
Hello!

Things I tested for:
1) Verified that the "+" button is present
2) Added two date and it did create two blue squares and 
3) Clicked on the blue square that was created and it showed created date 
Screenshot 2023-12-14 at 1 51 39 PM
Screenshot 2023-12-14 at 1 52 52 PM
Screenshot 2023-12-14 at 1 51 43 PM

**Comment by yuhuimin1996** [comment]
Hi Jiyuan,
I've tested this FE pr also with the BE pr #640. Everything works well as you described. After adding a blue square, by hovering on it, the create date will show up. And after clicking on this blue square, the created date will appear beside ‘Created Date:'. Nice job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108281570/b431ad50-8a52-46f6-bfa1-98e5b846b6e4

**Review by Sophie-lei — DISMISSED** [review]
hey! I just tested this PR, and everything works well. I tasted as admin role. After created 
 the blue square for other user with date and summary, the other user's blue square create date was shown as expected. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/70113822/0f234876-8405-41f6-a105-3df01f711b0f

**Review by pshereen — DISMISSED** [review]
I tested the PR as a admin user.

It works as per the requirement. 

Created date appears before the blue square description.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/7041610/5ea55f8a-6773-4222-b627-1fb06cfa1c25

**Review by Renanluizssx — DISMISSED** [review]
Hi @Jiyuanxia, I tested your application, and it's working very well. I navigated to the leaderboard, clicked on an admin user named Aaron, pressed the plus button to add a blue square, entered the date, and then submitted. It showed me the date when I hovered over the mouse and it show me a date 
 when i clicked on the blue square button.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/253631b9-0cea-43d6-85ab-2835592e58ad

**Review by X1aoZhang99 — DISMISSED** [review]
Hi Jiyuan, 

I've just tested this pull request, and everything is functioning correctly. I tested it in the admin role. After creating a blue square for another user with a date and summary, the creation date of the other user's blue square was displayed as anticipated.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113070229/98cde0e8-014e-43b2-b033-33c9be226349

**Review by Renanluizssx — DISMISSED** [review]

**Comment by Vishala09** [comment]
Works as intended! The created date appears now after creating a new blue square. But for some reason , I am getting a different date. Can you please let me know if this is expected?


[Untitled_ Jan 18,2024 4_31 PM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/65274029/e9a355a6-b881-4a59-b994-8ed42e0a4363)

Before with no created date:
image

After with created date:
image

**Review by Vishala09 — DISMISSED** [review]
Please check my previous comment for the issue.

**Comment by xmyoot** [comment]
Reviewed this PR and assigning blue squares functioned as expected, reference video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/15240538/aba637fb-f139-4055-adf5-afe5a8de7b2c

File: src/components/UserProfile/BlueSquares/BlueSquare.jsx [hunk 1]
```diff
@@ -15,37 +15,48 @@ const BlueSquare = (props) => {
...
+                    
+                    {blueSquare.description !== undefined && 
+                      <div className="summary">{blueSquare.createdDate !== undefined ? formatCreatedDate(BlueSquare.createdDate)+":"  : null}
+
+          
+                      {formatDateFromDescriptionString(blueSquare.description)}</div>
+                    }
+                  </div>
                 </div>
               </div>
```
Comment (rusalshrestha98) [inline src/components/UserProfile/BlueSquares/BlueSquare.jsx:61]: I think you may have an extra `</div` closing tag on line 61 that is causing the build to fail.

File: src/components/UserProfile/BlueSquares/BlueSquare.jsx [hunk 2]
```diff
@@ -15,37 +15,48 @@ const BlueSquare = (props) => {
...
-                    );
-                  }
-                }}
-              >
-                <div className="report" data-testid="report">
-                  <div className="title">{formatDate(blueSquare.date)}</div>
-                  {blueSquare.description !== undefined &&
-                    <div className="summary">{formatDateFromDescriptionString(blueSquare.description)}</div>
-                  }
+
```
Comment (rusalshrestha98) [inline src/components/UserProfile/BlueSquares/BlueSquare.jsx:18]: Might be better to remove this empty line space on line 18

File: src/components/UserProfile/BlueSquares/BlueSquare.jsx [hunk 3]
```diff
@@ -15,37 +15,48 @@ const BlueSquare = (props) => {
...
+                    
+                    </div>
+                    
+                    {blueSquare.description !== undefined && 
+                      <div className="summary">{blueSquare.createdDate !== undefined ? formatCreatedDate(BlueSquare.createdDate)+":"  : null}
+
+          
+                      {formatDateFromDescriptionString(blueSquare.description)}</div>
+                    }
+                  </div>
```
Comment (rusalshrestha98) [inline src/components/UserProfile/BlueSquares/BlueSquare.jsx:59]:
```suggestion
                  <div className="report" data-testid="report">
                    <div className="title">{formatDate(blueSquare.date)}</div>

                    {blueSquare.description !== undefined && (
                      <div className="summary">
                        {blueSquare.createdDate !== undefined
                          ? formatCreatedDate(BlueSquare.createdDate) + ':'
                          : null}

                        {formatDateFromDescriptionString(blueSquare.description)}
                      </div>
                    )}
                  </div>
```

Comment (rusalshrestha98) [reply src/components/UserProfile/BlueSquares/BlueSquare.jsx:59]: Try using Prettier extension in VS code to easily make your code look cleaner

**Review by rusalshrestha98 — DISMISSED** [review]
Please review above comments and resolve conflict in `formatDate.js`

**Comment by galgalinidhi** [comment]
Hi @Jiyuanxia, the FrontEnd doesn't compile because of the following error-

Screenshot 2024-01-20 at 7 44 33 PM

I commented out the extra <div> tag. After that, the code compiles, but I notice that the page is not responsive.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44816413/cb68157d-6847-48ee-a831-f79642ce3415

**Review by 666saofeng — CHANGES_REQUESTED** [review]
Hi, there is an extra <div> tag in BlueSquares.jsx. After I commented it out, the dashboard page became unresponsive.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62363348/f837f07a-ee9c-4f07-82c8-708222e33351

**Comment by Jiyuanxia** [comment]
Hi @rusalshrestha98 and @Vishala09 ala09, thank you for your previous review on my pull request. Unfortunately, I encountered some challenges while resolving conflicts, leading to a mistake. As a result, I've reverted the changes to a working version. Please review again. Thank you for understanding.

**Review by peterson337 — CHANGES_REQUESTED** [review]
I created the blue square, updated it, but the creation date is not displayed below the date entry.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73541319/85061512-2109-4417-9e0d-c38cc9a0dcfc

**Comment by Jiyuanxia** [comment]
> I created the blue square, updated it, but the creation date is not displayed below the date entry.
> 
>  Desktop.2024.01.22.-.12.28.40.04.mp4

Hi @peterson337 , the problem you mentioned should not happen. Check if you get the latest version and switch to the accourding backend PR.

**Comment by peterson337** [comment]
> > Criei o quadrado azul, atualizei-o, mas a data de criação não aparece abaixo da entrada de data.
> > Desktop.2024.01.22.-.12.28.40.04.mp4
> 
> Oi@peterson337, o problema que você mencionou não deveria acontecer. Verifique se você obteve a versão mais recente e mude para o PR de back-end correspondente.

[@Jiyuanxia](https://github.com/Jiyuanxia), Which branch are you on?

**Comment by peterson337** [comment]
> > I created the blue square, updated it, but the creation date is not displayed below the date entry.
> > Desktop.2024.01.22.-.12.28.40.04.mp4
> 
> Hi @peterson337 , the problem you mentioned should not happen. Check if you get the latest version and switch to the accourding backend PR.

Hi [@Jiyuanxia](https://github.com/Jiyuanxia), The changes you made have not been merged into the development branch. Which branch are you on?

**Comment by peterson337** [comment]
> > I created the blue square, updated it, but the creation date is not displayed below the date entry.
> > Desktop.2024.01.22.-.12.28.40.04.mp4
> 
> Hi @peterson337 , the problem you mentioned should not happen. Check if you get the latest version and switch to the accourding backend PR.
[@Jiyuanxia](https://github.com/Jiyuanxia), I was able to use your branch to test this PR, and it worked.

**Review by peterson337 — DISMISSED** [review]
I tested it, and everything works as expected.

one comunnity

**Review by SRamen1999 — DISMISSED** [review]
I've just tested this PR, and everything is functioning correctly. Both the admin and owner roles can create, edit, and delete blue squares. The date and summary is displayed without any issues.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/e73dde86-1299-443c-8729-53d7e068ee08

**Review by KevHnh — DISMISSED** [review]
Nice job, everything works as intended!

Using an owner account I was able to create, update, and delete a blue square. Everything functioned as planned and no issues were seen.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83934825/5f641241-24a6-4e06-9492-0506ea52bcfb

**Comment by xmyoot** [comment]
Hello, great work on this component, I'm getting this error on the front end, I've checked out to the corresponding branches on both front end and back end, npm installed and ensured node 14 was being used but I keep getting this error please reference the photo for more details: 
PR#640+1641

**Review by Pandani07 — APPROVED** [review]
Tested this PR and the related backend PR, and 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/00067703-5c23-4c87-b8fa-0339047d4b9f

everything works as expected. I could add a new blue square for a user. I could also check the created date for a blue square and update the description of an existing blue square.  The **created date** inside the blue square modal can be styled better but everything else is great!

**Pull request merged**